src/core/auth/qgsauthcertutils.cpp

```diff
@@ -262,37 +273,249 @@ QStringList QgsAuthCertUtils::certKeyBundleToPem( const QString &certpath,
   return QStringList() << certpem << keypem << algtype;
 }
 
+bool QgsAuthCertUtils::pemIsPkcs8( const QString &keyPemTxt )
```

Comment:
See my previous comment: can this function be used to detect PEM vs DER  instead of the file extension ?

Reply:
It doesn't do encoding determination. It specifically has to work with PEM in order to simply text search to determine PKCS format (much more involved if using DER). I suppose the signature could be more descriptive, e.g. `privateKeyPemIsPkcs8 (...)`.